MFVNeutralino/plugins/EventFilter.cc

```diff
@@ -6,119 +6,197 @@
+    if (jet_selector(jet)) {
+        ++njets;
+        if (jet.pt() > min_pt_for_ht)
+            ht += jet.pt();
     }
 
-    int njets = 0;
-    double ht = 0;
-    for (const pat::Jet& jet : *jets)
-        if (jet_selector(jet)) {
-            ++njets;
-            if (jet.pt() > min_pt_for_ht)
-                ht += jet.pt();
+  const bool jets_pass = njets >= min_njets && ht >= min_ht;
+
+  //leptons 
+  edm::Handle<pat::MuonCollection> muons;
+  edm::Handle<pat::ElectronCollection> electrons;
+  event.getByToken(muons_token, muons);
+  event.getByToken(electrons_token, electrons);
+  
+  edm::Handle<double> rho;
+  event.getByToken(rho_token, rho);
+
+  int nmuons = 0, nelectrons = 0;
+  bool leptons_pass = nmuons + nelectrons >= min_nleptons;
+  
+  for (const pat::Muon& muon : *muons) {
+    //if (muon_selector(muon) && muon.pt() > min_muon_pt)
+    reco::TrackRef mtk = muon.track();
+    if (mtk.isNull()){
+      continue;
+    }
+    if (muon.pt() > min_muon_pt && abs(muon.eta()) < 2.4) {
+      bool isMedMuon = muon.passed(reco::Muon::CutBasedIdMedium);
+      const float iso = (muon.pfIsolationR04().sumChargedHadronPt + std::max(0., muon.pfIsolationR04().sumNeutralHadronEt + muon.pfIsolationR04().sumPhotonEt -0.5*muon.pfIsolationR04().sumPUPt))/muon.pt();
+      if (isMedMuon && iso < 0.15) {
+  	    ++nmuons;
+      } 
+    } 
+    leptons_pass = nmuons + nelectrons >= min_nleptons;
+    if (leptons_pass)
+      break;
+  } 
+  if (leptons_pass && mode == Mode::muons_only ) return true; //avoid double counting an event that fires both lepton triggers and keep it in muon selection  
+  
+  if (mode == Mode::electrons_only_veto_muons){
+    if (leptons_pass){ //double-count event
+      leptons_pass = false;
+    }
+    else{
+      for (const pat::Electron& electron : *electrons) {
+        // if (electron_selector(electron) && electron.pt() > min_electron_pt)
+        //new electron selector : is cut based tight & iso < 0.10
+        reco::GsfTrackRef etk = electron.gsfTrack();
+        
+        if (etk.isNull()){
+          continue;
         }
-
-    const bool jets_pass = njets >= min_njets && ht >= min_ht;
-
-    if (debug) printf("MFVEventFilter: njets: %i  ht: %f pass? %i\n", njets, ht, jets_pass);
-
-    if (mode == Mode::jets_only) {
-        return jets_pass;
+        if (electron.pt() > min_electron_pt && abs(electron.eta()) < 2.4) {
+      
+          bool isTightEl = electron.electronID("cutBasedElectronID-Fall17-94X-V1-tight");
+          const auto pfIso = electron.pfIsolationVariables();
+          const float eA = electron_effective_areas.getEffectiveArea(fabs(electron.superCluster()->eta()));
+          const float iso = (pfIso.sumChargedHadronPt + std::max(0., pfIso.sumNeutralHadronEt + pfIso.sumPhotonEt - *rho*eA)) / electron.pt();
+          const bool passveto = electron.passConversionVeto();
+
+          if (isTightEl && passveto && iso < 0.10) {
+  	        ++nelectrons;
+          } 
```

Comment:
@pkotamnives and @abbywarden: Can we avoid hardcoding these electron and muon ID/isolation selections? I particularly worry because event filter isn't applied to e.g. signal samples, so if we have any mismatch here vs. what is in AnalysisCuts, we'll be applying different requirements to background and data than we do to signal! And we're applying Medium + isolation for muons here, and had Loose + no isolation in one of the places I left a comment on a little while ago, I think...?

Reply:
Hmm and we apply electron isolation here, which my earlier comment said we should check if the Twiki says it includes it in the ID WP already or not! :)